services/headless-lms/server/src/controllers/auth.rs

```diff
@@ -331,6 +382,56 @@ pub async fn post_new_user_to_moocfi(user_details: &CreateAccountDetails) -> any
     }
 }
 
+/// Puts updated user information to tmc.mooc.fi.
+pub async fn update_user_information_to_moocfi(
+    first_name: String,
+    last_name: String,
+    email: String,
+) -> anyhow::Result<()> {
+    let tmc_api_url = "https://tmc.mooc.fi/api/v8/users/current";
+    let access_token = env::var("TMC_ACCESS_TOKEN").expect("TMC_ACCESS_TOKEN must be defined");
+
+    let ratelimit_api_key = env::var("RATELIMIT_PROTECTION_SAFE_API_KEY")
+        .expect("RATELIMIT_PROTECTION_SAFE_API_KEY must be defined");
+
+    let tmc_client = Client::default();
+
+    let json = serde_json::json!({
+        "user": {
+            "email": email,
+        },
+        "user_field": {
+            "first_name": first_name,
+            "last_name": last_name
+        },
+    });
+
+    let res = tmc_client
```

Comment:
Create a new module in headless-lms-utils/services/tmc

Create TmcClient struct, that stores the access token, ratelimit key etc, then impl TmcClient -> functions for the different operations.

Reply:
https://github.com/rage/secret-project-331/blob/master/services/headless-lms/server/src/config.rs